tests/v2/Controller/BeneficiaryResetPasswordController/SecretAnswerTest.php

```diff
@@ -0,0 +1,145 @@
+    {
+        yield 'Should redirect to login when not authenticated' => [self::URL, 302, null, '/login'];
+        yield 'Should return 403 status code when authenticated as beneficiaire' => [self::URL, 403, BeneficiaryFixture::BENEFICIARY_MAIL];
+        yield 'Should return 200 status code when authenticated as member with relay in common' => [self::URL, 200, MemberFixture::MEMBER_MAIL_WITH_RELAYS_SHARED_WITH_BENEFICIARIES];
+        yield 'Should return 403 status code when authenticated as member with no relay in common' => [self::URL, 403, MemberFixture::MEMBER_MAIL];
+    }
+
+    /** @dataProvider provideTestRoute */
+    public function testRoute(
+        string $url,
+        int $expectedStatusCode,
+        ?string $userMail = null,
+        ?string $expectedRedirect = null,
+        string $method = 'GET',
+        bool $isXmlHttpRequest = false,
+        array $body = [],
+    ): void {
+        $beneficiary = BeneficiaireFactory::findByEmail(BeneficiaryFixture::BENEFICIARY_MAIL)->object();
+        $url = sprintf(self::URL, $beneficiary->getId());
+
+        $this->assertRoute($url, $expectedStatusCode, $userMail, $expectedRedirect, $method);
+    }
+
+    /**  @dataProvider provideTestFormIsValid */
+    public function testFormIsValid(string $url, string $formSubmit, array $values, ?string $email, ?string $redirectUrl): void
+    {
+        $beneficiary = BeneficiaireFactory::findByEmail(BeneficiaryFixture::BENEFICIARY_MAIL)->object();
+        $url = sprintf(self::URL, $beneficiary->getId());
+        $values['reset_password_secret_answer[secretAnswer]'] = $beneficiary->getReponseSecrete();
+
+        $this->assertFormIsValid($url, $formSubmit, $values, $email, $redirectUrl);
+    }
+
+    public function provideTestFormIsValid(): ?\Generator
+    {
+        $values = self::FORM_VALUES;
+        $values['reset_password_secret_answer[password][plainPassword][first]'] = 'sameCorrectPassword';
+        $values['reset_password_secret_answer[password][plainPassword][second]'] = 'sameCorrectPassword';
+
+        yield 'Should redirect to beneficiaries list when form is correct' => [
+            self::URL,
+            'confirm',
+            $values,
+            MemberFixture::MEMBER_MAIL_WITH_RELAYS_SHARED_WITH_BENEFICIARIES,
+            '/professional/beneficiaries',
+        ];
+    }
+
+    /**
+     * @dataProvider provideTestFormIsNotValid
+     */
+    public function testFormIsNotValid(string $url, string $route, string $formSubmit, array $values, array $errors, ?string $email, ?string $alternateSelector = null): void
+    {
+        $beneficiary = BeneficiaireFactory::findByEmail(BeneficiaryFixture::BENEFICIARY_MAIL)->object();
+        $url = sprintf(self::URL, $beneficiary->getId());
+
+        if ($values['reset_password_secret_answer[secretAnswer]'] === 'goodSecretAnswer') {
+            $values['reset_password_secret_answer[secretAnswer]'] = $beneficiary->getReponseSecrete();
+        }
```

Comment:
Actually you don't need to do this. We use a BeneficiaryFactory that declares a default value to reponseSecrete. You could just pass the right value directly in the data provider insted of 'goodSecretAnswer'